### PR TITLE
Add publicEndpoint to override resolved endpoint

### PR DESCRIPTION
**Pull request by ktonon**
Trying to get everything using https. This PR makes it possible to serve the app resolver using ngrok

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -78,6 +80,7 @@ var target = appResolver.getUrl();
   - `dist` - The directory containing the app files to serve.  By default, the `dist` directory is used.
   - `port` - The port to listen on.  By default, port `3000` is used, which is the port that the LMS expects it on.
   - `hostname` - The hostname (or IP) to listen on. By default, the hostname of the operating system is used.  You should not need to change this.
+  - `publicEndpoint` - If provided overrides the protocol (http) hostname and port for endpoint resolution
```
Comment (dlockhart) [inline README.md:80]: Another option would be to make the new parameter `protocol`, so then you could use the existing `port` and `hostname` options and just set `protocol` = `https`. 🤷

Comment (ktonon) [reply README.md:80]: I tried that first, but I ran into trouble when trying to run the server. I also had to change the value of the `port`, which messed up this line:

```js
app.listen(self._opts.port, function() {
```

I only want to override the value which gets written into the `appconfig.json` file, i.e. for the loader endpoint. Having a single value to override makes it easier to customize the endpoint from the command line.

For example, in the smart-curriculum project, I'm doing this:

```sh
npm run build:config -- --publicEndpoint=https://arkt.ngrok.io
```

Comment (dlockhart) [reply README.md:80]: Cool, makes sense!

**Review by dlockhart — APPROVED** [review]

**Pull request merged**